optuna/__init__.py

```diff
@@ -20,3 +22,37 @@
 from optuna.study import Study  # NOQA
 from optuna.trial import Trial  # NOQA
 from optuna.version import __version__  # NOQA
+from optuna.type_checking import TYPE_CHECKING  # NOQA
+
+
+if TYPE_CHECKING:
+    from optuna import dashboard  # NOQA
+else:
+    from typing import Any
+
+    class _LazyImport(types.ModuleType):
+        """Module Wrapper for lazy import.
+
+        This class wrap specified module and lazily import it when they are actually accessed.
+        Otherwise, `import optuna` becomes slower because it imports all submodules and
+        their dependencies (e.g., bokeh) all at once.
+        Within this project's usage, importlib override this module's attribute on the first
+        access and the imported submodule is directly accessed from the second access.
+
+        Args:
+            name: name of module to apply lazy import
```

Comment:
Just to be consistent with other classes in Optuna, please start the description with a capital and end with a period.
```suggestion
            name: Name of module to apply lazy import.
```